main.cpp

```diff
@@ -41,22 +41,458 @@ If you need inspiration for what to write, take a look at previously approved st
+
+    if(amountPerIndividual <= 2)
+        cout << "they can have more tea" << endl;
+    else
+    {
+        tooMuchTea = true;
+        cout << "too much tea" << endl;
+    }
+    if (food)
+        cout << "they should eat something" << endl;
+    return tooMuchTea;
+}
+
+void TeaParty::spill()
+{
+    auto amountSpilled = (totalNumberOfCupsServed / participants);
+    for(int i = 0; i < amountSpilled; i++)
+    {
+        cout << " 1 cup is spilled " << endl;
+    }
+}
 
 /*
  new UDT 4:
  */
+ struct PastaShopNewHire
+ {
+    PastaShopNewHire()
+    {
+        cout << "construct PastaShopNewHire" << endl;
+    }
+
+    ~PastaShopNewHire()
+    {
+        cout << "destruct PastaShopNewHire" << endl;
+    }
+
+    PastaShop pastaShop;
+
+    void sellNoodles(int soldAmount);
+    int packNoodles(int packages_);
+    int output();
+
+    int packages;
+    int customShapesDesigned;
+    int packagingPay;
+    int moneyMade;
+    double energy {1.0f};
+ };
+
+void PastaShopNewHire::sellNoodles(int soldAmount)
+{
+    for(int i = 0; i < soldAmount; i++)
+    {
+        pastaShop.makeCustomPasta(1.3, "New Shape", true, true, 1.0);
+    }
+}
+
+int PastaShopNewHire::packNoodles(int packages_)
+{
+    packages = packages_;
+    packagingPay = static_cast<int>(packages_ * pastaShop.pastaPrice / 3);
+    return packagingPay;
+}
+
+int PastaShopNewHire::output()
+{
+    moneyMade = pastaShop.pastaProfit + packagingPay;
+
+    for(int i = static_cast<int>(pastaShop.amountOfDoughlBS); i > 10; i--)
+    {
+       energy = energy - 0.1;
+       cout << "total energy to make pasta left " << energy << endl;
+    }
+    return moneyMade;
+}
 
 /*
  new UDT 5:
  */
+struct TrainRide
+{
+    TrainRide()
+    {
+        cout << "construct TrainRide" << endl;
+    }
+
+    ~TrainRide()
+    {
+        cout << "destruct TrainRide" << endl;
+    }
+
+    bool awake;
+    bool rain;
+    int timeAsleep; 
+    double distanceTravelled = 0.0;
+    double trainSpeedPerHour = 30.0;
+
+    double progressMade(double distance_, double trainSpeed_, int time_);
+    bool wakeUp();
+    bool goToSleep();
+
+    TeaParty teaParty{15};
+};
+
+double TrainRide::progressMade(double distance_, double trainSpeed_, int time_)
+{
+    if((distance_ > 0) && (trainSpeed_ = 0.0))
+    {
+        goToSleep();
+    }
+
+    if(distance_ < (trainSpeed_ * time_))
+        cout << "will get to the destination on time " << endl;
+    else
+       cout << "will not get to the destination on time " << endl; 
+
+    distanceTravelled = distanceTravelled + distance_;
+    return distanceTravelled;
+
+}
+
+bool TrainRide::wakeUp()
+{
+    if(!awake)
+    {
+        teaParty.drink(1, 1, true);
+        cout << "wake up " << endl;
+    }
+    else if(!awake && rain)
+    {
+        teaParty.drink(1, 2, true);
+        cout << "wake up with 2 cups" << endl;
+    }
+    else
+        cout << "awake" << endl;
+    return awake = true;
```

Comment:
curly braces
`return awake = true;` is weird.   are you assigning `true` to `awake`? and then returning the value of `awake`? 
What are you trying to do express here with this `return` statement?   It's impossible to tell what your intention is here.

- [ ] revise